### ISSUE TITLE
Mach-O file parsing missing support for INDIRECT_SYMBOL_LOCAL, INDIRECT_SYMBOL_ABS

### ISSUE THREAD

**Issue by xusheng6** [Vector35/binaryninja-api#3421]
While analyzing the file [virtual-test.zip](https://github.com/Vector35/binaryninja-api/files/9388797/virtual-test.zip), when I view the file statically, I get the following value at address 0x100004060:

<img width="511" alt="Screen Shot 2022-08-21 at 5 32 02 PM" src="https://user-images.githubusercontent.com/94503187/185784918-2c8e88a9-3ad6-426c-a33f-03f1839cc6e8.png">

The value 0x100000000040d0 is wrong here -- it is a vtable, but it is beyond the address range of the current binary view. If I launch it under the debugger, I can see its correct value:

<img width="511" alt="Screen Shot 2022-08-21 at 5 32 29 PM" src="https://user-images.githubusercontent.com/94503187/185784946-754143ec-6d16-4828-98d9-d7c8b638af0c.png">

And we can see its content:

<img width="619" alt="Screen Shot 2022-08-21 at 5 38 08 PM" src="https://user-images.githubusercontent.com/94503187/185784977-2ec3dd2d-7e56-41a8-8d22-28f3e42f5240.png">

I have no idea why it is happening, but I suspect this is related to relocation. Btw, this is an arm64 file so it could also be related

**Comment by plafosse** [comment]
I would suspect relocations

**Comment by plafosse** [comment]
So this does have something to do with relocations but not really. For some reason in the binary the table of `pointerSymbols` has `0x80000000` as the index

![image](https://user-images.githubusercontent.com/10556585/187456452-0503f9ca-44d7-4f25-ac15-d50526cef12c.png)

This must have some significance that we're unaware of.

```
		unordered_map<size_t, vector<std::pair<section_64*, size_t>>> pointerSymbols;
		for (auto& symbolPointerSection : m_symbolPointerSections)
		{
			size_t needed = symbolPointerSection.size / m_addressSize;
			for (size_t j = 0; (j < needed) && ((j + symbolPointerSection.reserved1) < indirectSymbols.size()); j++)
			{
				LogError("pointerSymbols %d %d %d", j, j+symbolPointerSection.reserved1, indirectSymbols[j + symbolPointerSection.reserved1]);
				pointerSymbols[indirectSymbols[j + symbolPointerSection.reserved1]].push_back(
					std::pair<section_64*, size_t>(&symbolPointerSection, j));
			}
		}
```

Adding in the above logging code gives us:

```
pointerSymbols 0 10 4
pointerSymbols 1 11 1
pointerSymbols 2 12 5
pointerSymbols 3 13 6
pointerSymbols 4 14 7
pointerSymbols 5 15 8
pointerSymbols 6 16 9
pointerSymbols 7 17 10
pointerSymbols 8 18 2
pointerSymbols 9 19 11
pointerSymbols 10 20 12
pointerSymbols 11 21 13
pointerSymbols 12 22 -2147483648
pointerSymbols 13 23 -2147483648
pointerSymbols 14 24 -2147483648
pointerSymbols 15 25 16
pointerSymbols 16 26 17
```

**Comment by plafosse** [comment]
Looked into this a bit more looks like we're missing support for `INDIRECT_SYMBOL_LOCAL` and `INDIRECT_SYMBOL_ABS`
 More info here:
https://opensource.apple.com/source/xnu/xnu-792/EXTERNAL_HEADERS/mach-o/loader.h.auto.html

**Comment by plafosse** [comment]
In current dev (soon to be stable) builds you at least get notice when we fail to parse these types of symbols.

**Comment by cxnder** [comment]
The bad pointer values are chained fixups. ( #2867 ) 
![image](https://user-images.githubusercontent.com/55725881/199750659-21b2f5d5-1ce7-4be4-b372-3b19d3875c01.png)

**Comment by cxnder** [comment]
Resolved in builds >= 3.2.3871
